pkg/config/generate.go

```diff
@@ -219,7 +221,7 @@ func Generate(cs *acsapi.OpenShiftManagedCluster) (err error) {
 			extKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 			key:         &c.Certificates.OpenshiftConsole.Key,
 			cert:        &c.Certificates.OpenshiftConsole.Cert,
-			selfSign:    true,
+			selfSign:    cs.Properties.FQDN != cs.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname,
```

Comment:
for now, how about always sign it with CA? entirely remove the selfSign pieces here and place a comment in the code that when we allow the user to specify this cert, remember issue 307 and they can only specify a cert if PublicHostname != FQDN